Added support for 'application/x-www-form-urlencoded'

ShibaniMahapatra opened the pull request:
Currently this library supports only 'application/json' as content-type, This PR aims in adding support to the other content-type supported by Github for webhooks i.e. 'application/x-www-form-urlencoded'.

This support makes python-github-webhook fully compatible with the github webhooks.

Signed-Off-By: Shibani Mahapatra <shibani.mahapatra47@gmail.com>

ShibaniMahapatra commented:
@alexchamberlain Please review this PR, and let know if any changes are required.

alexchamberlain reviewed (commented):
Hi @ShibaniMahapatra, thanks for your contribution. Sorry for the delay; I've had some personal commitments running up to Christmas.

For context, what is the reason you specifically want the form encoding to be supported over the JSON format?

To land this, please can you add a test that uses form encoded data?

ShibaniMahapatra commented:
@alexchamberlain 
The reason for supporting this content-type is to provide support for users who are already using 'application/x-www-form-urlencoded' as the content-type instead of 'application/json', so that they can also seamlessly integrate with github webhooks with this library. 

Since _github supports two content-type_, it will be good to provide support for those two types. _Earlier this library was supporting only 'application/json'._
So finally this library will support **both 'application/json' and 'application/x-www-form-urlencoded'** as content-type.

Also, I have committed testcases for the 'application/x-www-form-urlencoded' content-type.
Please review it. Thanks.

ShibaniMahapatra committed:
Add support for 'application/x-www-form-urlencoded'

Add support for Content-Type: application/x-www-form-urlencoded, so
that `python-github-webhook` supports both Content-Types supported
by GitHub webhooks.

Signed-Off-By: Shibani Mahapatra <shibani.mahapatra47@gmail.com>

alexchamberlain commented:
Released as 1.0.3; @ShibaniMahapatra Thank You for your contribution; really appreciate it!

ShibaniMahapatra commented:
Thank you @alexchamberlain. Looking forward to many such contributions. 👍
